equinox/nn/_attention.py

```diff
@@ -297,10 +375,8 @@ def __call__(
             )
 
         query_seq_length, _ = query.shape
-        kv_seq_length, _ = key_.shape
-        kv_seq_length2, _ = value.shape
-        if kv_seq_length != kv_seq_length2:
-            # query length can be different
+
+        if (kv_seq_length := key_.shape[0]) != value.shape[0]:
```

Comment:
Actually, one benefit of the previous approach was that it implicitly checked the number of dimensions in the input as well. I think that's important!